server/data_common/data_adaptor.py

```diff
@@ -400,7 +400,15 @@ def layout_to_fbs_matrix(self, fields):
         with ServerTiming.time("layout.query"):
             for ename in embeddings:
                 embedding = self.get_embedding_array(ename, 2)
-                normalized_layout = DataAdaptor.normalize_embedding(embedding)
+                # If this is a spatial dataset, the scaling needs to be to pixel space, not point space.
+                print("NAME IS WHAT: ", ename)
+                if ename == "spatial":
+                    print(self.get_spatial_extent())
+                    normalized_layout = embedding / self.get_spatial_extent()
+                    
+                    normalized_layout = normalized_layout.astype(dtype=np.float32)
```

Comment:
might(!) have been nicer to call `normalize_embedding` with another parameter. But definitely good enough